src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/AddUsingsCSharpCodeActionResolver.cs

```diff
@@ -46,9 +49,20 @@ internal class AddUsingsCSharpCodeActionResolver : CSharpCodeActionResolver
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            if (!AddUsingsCodeActionProviderHelper.TryExtractNamespace(codeAction.Title, out var @namespace))
+            var resolvedCodeAction = await ResolveCodeActionWithServerAsync(csharpParams.RazorFileUri, codeAction, cancellationToken).ConfigureAwait(false);
+
+            // TODO: Move this higher, so it happens on any code action.
+            //       For that though, we need a deeper understanding of applying workspace edits to documents, rather than
+            //       just picking out the first one because we assume thats where it will be.
+            //       Tracked by https://github.com/dotnet/razor-tooling/issues/6159
+            if (resolvedCodeAction?.Edit?.TryGetDocumentChanges(out var documentChanges) != true)
+            {
+                return codeAction;
+            }
+
+            if (documentChanges!.Length != 1)
             {
-                // Invalid text edit, missing namespace
+                // We don't yet support multi-document code actions, return original code action
```

Comment:
Hmm, so in the multi-document case we'll display a code action but it'll do nothing? In that case should we prevent the code action from showing up in the first place?

Reply:
A very good question! The Add Usings code action from Roslyn doesn't ever return multiple document edits. Unfortunately if that does change (and maybe i'll throw a debug assert in here just in case) then we can't tell at code action providing time, we'd have to just block the add usings code action, which would require a code change anyway.

In an ideal world this whole block of code would move to a higher level, as per the TODO comment above), so I put this here so that hopefully nobody is tempted to think that is too straight forward :)